### PR TITLE
Replace arangodb::containers::SmallVector with std::span and absl::InlinedVector

### PR DESCRIPTION
**Pull request by MBkkt**
### Scope & Purpose

Replace our own primitives with well tested (also nice performance) from 3rdParty library.
Also I think `std::span<T const>` probably always better then `SomeContainer<T> const&`

- [ ] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [x] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.9: *(Please link PR)*
  - [ ] Backport for 3.8: *(Please link PR)*
  - [ ] Backport for 3.7: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR: https://github.com/arangodb/enterprise/pull/936
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: arangod/RocksDBEngine/RocksDBVPackIndex.h [hunk 1]
```diff
@@ -197,32 +199,31 @@ class RocksDBVPackIndex : public RocksDBIndex {
...
 
   /// @brief helper function to insert a document into any index type
-  ErrorCode fillElement(
-      velocypack::Builder& leased, LocalDocumentId const& documentId,
-      VPackSlice doc, ::arangodb::containers::SmallVector<RocksDBKey>& elements,
-      ::arangodb::containers::SmallVector<uint64_t>& hashes);
+  ErrorCode fillElement(velocypack::Builder& leased,
+                        LocalDocumentId const& documentId, VPackSlice doc,
+                        absl::InlinedVector<RocksDBKey, 4>& elements,
+                        absl::InlinedVector<uint64_t, 4>& hashes);
```
Comment (jsteemann) [inline arangod/RocksDBEngine/RocksDBVPackIndex.h:205]: For a faithful port of the old code, I think this should have 8 elements. But as the other out variable has only capacity for 4 elements, it is absolutely fine :+1:

File: arangod/Transaction/Context.h [hunk 2]
```diff
@@ -134,10 +134,8 @@ class Context {
   TRI_vocbase_t& _vocbase;
   std::unique_ptr<velocypack::CustomTypeHandler> _customTypeHandler;
 
-  ::arangodb::containers::SmallVectorWithArena<arangodb::velocypack::Builder*,
-                                               32>
-      _builders;
-  ::arangodb::containers::SmallVectorWithArena<std::string*, 32> _strings;
+  absl::InlinedVector<arangodb::velocypack::Builder*, 4> _builders;
```
Comment (jsteemann) [inline arangod/Transaction/Context.h:137]: As we are handling out `Builder` objects often, I suggest we increase the capacity to `8` here.
```suggestion
  absl::InlinedVector<arangodb::velocypack::Builder*, 8> _builders;
```

File: tests/Aql/BitFunctionsTest.cpp [hunk 3]
```diff
@@ -46,8 +46,7 @@ namespace {
 
 AqlValue callFn(AstNode const& node, char const* input1,
                 char const* input2 = nullptr, char const* input3 = nullptr) {
-  SmallVector<AqlValue>::allocator_type::arena_type arena;
-  SmallVector<AqlValue> params{arena};
+  absl::InlinedVector<AqlValue, 3> params;
```
Comment (jsteemann) [inline tests/Aql/BitFunctionsTest.cpp:49]: :+1:

File: arangod/Aql/OptimizerRules.cpp [hunk 4]
```diff
@@ -8272,11 +8229,11 @@ void arangodb::aql::activateCallstackSplit(ExecutionPlan& plan) {
 namespace {
 
 void findSubqueriesSuitableForSplicing(
-    ExecutionPlan const& plan, containers::SmallVector<SubqueryNode*>& result) {
+    ExecutionPlan const& plan, absl::InlinedVector<SubqueryNode*, 8>& result) {
   TRI_ASSERT(result.empty());
   using ResultVector = decltype(result);
   using BoolVec =
-      std::vector<bool,
+      std::vector<bool,  // TODO(MBkkt) vector<bool>? Really?
```
Comment (jsteemann) [inline arangod/Aql/OptimizerRules.cpp:8236]: Let's replace it with a vector of `uint8_t`s...!

**Review by jsteemann — APPROVED** [review]
LGTM!

File: arangod/Aql/ConstFetcher.cpp [hunk 5]
```diff
@@ -29,7 +29,6 @@
 #include "Aql/SkipResult.h"
 #include "Basics/Exceptions.h"
 #include "Basics/voc-errors.h"
-#include "Containers/SmallVector.h"
```
Comment (jsteemann) [inline arangod/Aql/ConstFetcher.cpp:32]: Shouldn't this include be kept now, given that SmallVector is used in this file but is not included in the header anymore?

**Review by jsteemann — APPROVED** [review]
LGTM!

**Review by gnusi — APPROVED** [review]

**Pull request merged**